server/src/main/java/org/elasticsearch/cluster/routing/ExpectedShardSizeEstimator.java

```diff
@@ -48,7 +48,7 @@ public static long getExpectedShardSize(
             && shard.active() == false
             && shard.recoverySource().getType() == RecoverySource.Type.LOCAL_SHARDS) {
             return getExpectedSizeOfResizedShard(shard, defaultValue, indexMetadata, clusterInfo, metadata, routingTable);
-        } else if (shard.unassigned() && shard.recoverySource().getType() == RecoverySource.Type.SNAPSHOT) {
+        } else if (shard.primary() && shard.active() == false && shard.recoverySource().getType() == RecoverySource.Type.SNAPSHOT) {
             return snapshotShardSizeInfo.getShardSize(shard, defaultValue);
```

Comment:
This is changed to align with code in `SnapshotShardSizeInfo#getShardSize`
https://github.com/elastic/elasticsearch/blob/e1c284a7c3aac0a238e8ac5fa4fe8a29f33e5685/server/src/main/java/org/elasticsearch/snapshots/SnapshotShardSizeInfo.java#L27-L29